zipline/pipeline/term.py

```diff
@@ -516,6 +585,31 @@ def postprocess(self, data):
         """
         return data
 
+    def _downsampled_type(self):
+        """
+        The expression type to return from self.downsample().
+        """
+        raise NotImplementedError(
+            "downsampling is not yet implemented "
+            "for instances of %s." % type(self).__name__
+        )
+
+    def downsample(self, frequency):
+        """
+        Make a term that computes from ``self`` at lower-than-daily frequency.
+
+        Parameters
+        ----------
+        frequency : str, {'Y', 'Q', 'M', 'W'}
+            A string indicating the desired sampling rate.
+            'Y' -> sample on the first trading day of each calendar year
```

Comment:
I agree that we should not make it close but slightly different to pandas. If you really don't like the pandas version then I think the verbose names are fine. I think people will look this up when they need to use it anyways so following the pandas convention can help the few people that do know them.